includes/Core/Modules/Modules.php

```diff
@@ -1018,4 +1018,36 @@ public function get_shareable_modules_owners() {
 	public function delete_dashboard_sharing_settings() {
 		return $this->options->delete( Module_Sharing_Settings::OPTION );
 	}
+
+	/**
+	 * Gets the connected modules.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Connected modules as $slug => $module pairs.
+	 */
+	public function get_connected_modules() {
+		$modules = $this->get_available_modules();
+
+		return array_filter(
+			$modules,
+			function( Module $module ) {
+				return $this->is_module_connected( $module->slug );
+			}
+		);
+	}
+
+	/**
+	 * Checks whether the module identified by the given slug is shareable.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug Unique module slug.
+	 * @return bool True if module is shareable, false otherwise.
+	 */
+	public function is_module_shareable( $slug ) {
+		$modules = $this->get_shareable_modules();
+
+		return isset( $modules[ $slug ] );
+	}
```

Comment:
As a solution to the above, we are introducing the `Modules::is_module_shareable` method, which checks if the module belongs to the array returned by the `Modules::get_shareable_modules()` method.

The `Modues::get_shareable_modules()` still uses `$module->is_shareable()`, but only adds it to the array being returned if the said module is connected.